modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/RoleResourceImpl.java

```diff
@@ -85,6 +85,21 @@ public void deleteOrganizationRoleUserAccountAssociation(
 			userAccountId, organization.getGroupId(), new long[] {roleId});
 	}
 
+	@Override
+	public void deleteRole(Long roleId) throws Exception {
+		_roleService.deleteRole(roleId);
+	}
+
+	@Override
+	public void deleteRoleByExternalReferenceCode(String externalReferenceCode)
+		throws Exception {
+
+		com.liferay.portal.kernel.model.Role role = _getServiceBuilderRole(
```

Comment:
I think we can remove the _getServiceBuilderRole method because we can use directly the _roleService.getRoleByExternalReferenceCode

Reply:
@smottal I did create here, as well the getRoleByERC method in service layer and substituted all the _getServiceBuilderRole() calls.